### PR TITLE
Remove fused_log_softmax option from RNNT Loss

### PR DESCRIPTION
**Pull request by carolineechen**
Note: merge after #1610

Remove `fused_log_softmax` option and keep the behavior of the default `True` option. `fused_log_softmax=False` option was only relevant when `reuse_logits_for_grads=True`, but this behavior is no longer supported after #1610.

### CODE REVIEW

**Review by vincentqb — APPROVED** [review]
LGTM
